app/questionnaire/questionnaire_store_updater.py

```diff
@@ -496,7 +496,14 @@ def remove_dependent_blocks_and_capture_dependent_sections(self) -> None:
             section_key,
             blocks_to_remove,
         ) in self.dependent_block_id_by_section_key.items():
-            if section_key not in self.started_section_keys():
+            # We don't want to skip current iteration if the block has dynamic answers,
+            # if block has dynamic answers we don't use list_item_id from section_key set, so the location is removed in
+            # self.remove_completed_location and dependencies updated
+            if section_key not in self.started_section_keys() and not any(
+                block_id
+                for block_id in blocks_to_remove
+                if self._schema.block_has_dynamic_answer(block_id)
```

Comment:
Was thinking rather than doing this check on every block we could add a method to questionnaire schemas to populate a set of block_ids containing dynamic answers upfront and use that here and below instead?

Reply:
I guess, it makes sense. We invoke it 3x so far.

Reply:
Added now, thanks!